### PR TITLE
Merge 2024.09-rc branch into develop

### PR DESCRIPTION
**Pull request by Art4**
The `2024.09-rc` branch is multiple commits away from the `develop` branch. This leads to PRs, where we have to wait for a release, before they could be merge, like #14529. This should be considered as tecnical dept. To quote the [refactoring.guru](https://refactoring.guru/refactoring/technical-debt) as source:

> ## Causes of technical debt
> 
> **Long-term simultaneous development in several branches**
> 
> This can lead to the accumulation of technical debt, which is then increased when changes are merged. The more changes made in isolation, the greater the total technical debt.

Waiting for a release hinders the development. Even if we wait for the release every open PR has the potential for new git conflicts that has to be fixed in every PR. This should not be an issue.

Please see also the [famous git flow by Vincent Driessen](https://nvie.com/posts/a-successful-git-branching-model/), where commits to a RC branch should be continuous merge back into the develop branch.

This PR merges the `2024.09-rc` branch into the `develop` branch, so I can fix more potential errors reported by PHPStan level 0. I recommend to port every commit into a `rc` branch directly to the `develop` branch or at least once in a week.

### CODE REVIEW

File: VERSION
Comment (nupplaphil) [inline VERSION:1]: Plz beware of the version-code mixups. :)
f.e. docker has some upgrade-checks based on it.

**Review by nupplaphil — CHANGES_REQUESTED** [review]

File: VERSION
Comment (Art4) [reply VERSION:1]: Thank you :+1: Yes, I'm aware that there are version mixups, but I'm not sure atm how they have to been solved. Should this be `2024.12-dev` now? (Another reason why there should no isolated development branches.)

File: src/App.php [hunk 1]
```diff
@@ -51,7 +51,7 @@ class App
 {
 	const PLATFORM = 'Friendica';
 	const CODENAME = 'Yellow Archangel';
-	const VERSION  = '2024.09-dev';
+	const VERSION  = '2024.09-rc';
```
Comment (Art4) [inline src/App.php]: ?
```suggestion
	const VERSION  = '2024.12-dev';
```

File: VERSION [hunk 2]
```diff
@@ -1 +1 @@
-2024.09-dev
+2024.09-rc
```
Comment (Art4) [inline VERSION]: ?
```suggestion
2024.12-dev
```

File: database.sql [hunk 3]
```diff
@@ -1,6 +1,6 @@
 -- ------------------------------------------
--- Friendica 2024.09-dev (Yellow Archangel)
--- DB_UPDATE_VERSION 1573
+-- Friendica 2024.09-rc (Yellow Archangel)
```
Comment (Art4) [inline database.sql]: ?
```suggestion
-- Friendica 2024.12-dev (Yellow Archangel)
```

**Comment by MrPetovan** [comment]
The theory behind the RC branch is that during the RC period, only fixes should be committed to the RC branch, while new features/enhancements should be committed for the next release on the develop branch. After the release, the develop branch is brought up to speed with the RC fixes and development carries on.

In practice, since @annando is the only main developer on Friendica, the RC period can stretch longer than the expected 1 month, and features/enhancements end up making their way in the RC branch as well. However, I don't think changing this workflow for anything more involved would be warranted. We're already not complying very well with our simple structure, so we probably wouldn't gain more from adding more complexity to the structure.

> Please see also the [famous git flow by Vincent Driessen](https://nvie.com/posts/a-successful-git-branching-model/), where commits to a RC branch should be continuous merge back into the develop branch.

I found the RC branch workflow of this flow very similar to ours, with a merge back to develop at the end of the release period, with no "continuous" merge back that I could see.

**Comment by Art4** [comment]
> The theory behind the RC branch is that during the RC period, only fixes should be committed to the RC branch, while new features/enhancements should be committed for the next release on the develop branch. After the release, the develop branch is brought up to speed with the RC fixes and development carries on.
> 
> In practice, since @annando is the only main developer on Friendica, the RC period can stretch longer than the expected 1 month, and features/enhancements end up making their way in the RC branch as well. However, I don't think changing this workflow for anything more involved would be warranted. We're already not complying very well with our simple structure, so we probably wouldn't gain more from adding more complexity to the structure.

@MrPetovan All of this is fine. I see no need to change the workflow for RC branches. By the way is this the nearly exact same workflow as in git flow described.

> > Please see also the [famous git flow by Vincent Driessen](https://nvie.com/posts/a-successful-git-branching-model/), where commits to a RC branch should be continuous merge back into the develop branch.
> 
> I found the RC branch workflow of this flow very similar to ours, with a merge back to develop at the end of the release period, with no "continuous" merge back that I could see.

Yes, and the missing continuous merges back to develop is the only thing I recommend to change. :blush: I am willing to take on this task. I only need some advise If the version naming is correct 

I bumped the version to `2024.12-dev` as seen in https://github.com/friendica/friendica/commit/60ffd8fe7882ad0ec315accd6102df30a07a4cc2 and I merged the newest changes from the `develop` branch and fixed the conflicts.

**Review by nupplaphil — APPROVED** [review]

**Review by MrPetovan — APPROVED** [review]

**Pull request merged**